website/content/docs/job-specification/group.mdx

```diff
@@ -329,6 +329,46 @@ group "second" {
 }
 ```
 
+#### Known Limitations
+
+The `max_client_disconnect` configuration is a new operating model for Nomad. Due
+to the nature of the change and the challenges it does and does not address, there
+are some known limitations to the feature that are detailed below.
+
+##### Minimum Server Version
+
+This feature introduces new node and allocation statuses as well as scheduling
+behavior that older versions of Nomad are unaware of. Given those constraints,
+it is not possible to leverage the `max_client_disconnect` configuration in clusters
+with servers running older versions of Nomad. The implementation includes guard
+logic that guarantees all servers in the cluster are running at least version
+1.3.0. If all servers in the cluster do not meet the minimum required version,
+Nomad will operate as it did prior to 1.3.0. The `max_client_disconnect`
+configuration will be _ignored_.
+
+##### Mixed Client Versions
+
+Similar to servers, clients running older versions of Nomad are unaware of the
+new node and allocation statuses. Unlike servers, there is currently no guard
+logic that checks all client versions during the scheduling process. The performance
+impact of such a check during scheduling is prohibitive. Attempting to leverage
+the `max_client_disconnect` feature in clusters with mixed client versions is _not
+supported_. Operators must ensure all clients meet the 1.3.0 minimum version requirement
+prior to deploying jobs with this configuration. If a job with this configuration
+is deployed to a cluster that does not meet this requirement, operators should
+run `nomad job stop -purge` on the job and remove the configuration before re-deploying.
```

Comment:
Most of this kind of stuff applies to _every_ new feature, and it's very wordy, and none of it is going to change over time. We have a warning block syntax (`~> **Note:** `), so I'd just wrap this all up like so:

```suggestion
~> **Note:** The `max_client_disconnect` feature is only supported on Nomad 
version 1.3.0 and above. If you run a job with `max_client_disconnect` on servers 
where some servers are not upgraded to 1.3.0, the `max_client_disconnect` 
flag will be _ignored_. Deploying a job with `max_client_disconnect` to a 
`datacenter` of Nomad clients where all clients are not 1.3.0 or above is unsupported.
```